tests/test_trace.py

```diff
@@ -210,3 +213,71 @@ def f2():
 
     assert recorder.spans[-1].parent_id is None
     assert recorder.spans[-1].operation_name == 'test_trace'
+
+
+def test_trace_single_with_tracer_args():
+
+    tags = {'t1': 'v1'}
+    operation_name = 'op_name'
+    component = 'component'
+
+    @trace(tags=tags, operation_name=operation_name, component=component)
+    def f1():
+        pass
+
+    recorder = Recorder()
+    opentracing.tracer = BasicTracer(recorder=recorder)
+
+    test_span = opentracing.tracer.start_span(operation_name='test_trace')
+
+    with test_span:
+        f1()
+
+    recorder.spans[0].tags == tags.update({opentracing_tags.COMPONENT: component})
+
+
+@pytest.mark.parametrize('return_span', (True, False))
+def test_trace_single_with_extractor(return_span):
+    recorder = Recorder()
+    opentracing.tracer = BasicTracer(recorder=recorder)
+
+    test_span = opentracing.tracer.start_span(operation_name='test_trace')
+
+    other_span = opentracing.tracer.start_span(operation_name='other_span')
+
+    def extractor():
+        return test_span if return_span else None
+
+    @trace(span_extractor=extractor)
+    def f1():
+        pass
+
+    with other_span:
+        # other_span could be ignored if extractor returned a span!
+        f1()
+
+    if return_span:
+        recorder.spans[0].context.trace_id == test_span.context.trace_id
```

Comment:
do you need `assert` for these statements?

Reply:
yes. nice catch *facepalm*